text/0027-foreman-plugin-forge.md

```diff
@@ -0,0 +1,141 @@
+README.md file in the project root.
+
+Keeping information up to date on each page is a manual task and could use some
+automation.
+
+A new Forge would address the following points:
+
+* Showcasing Foreman plugin ecosystem and discover new plugins
+* Offer helpful commands/code snippet for installing the plugin via source, gem,
+  rpm, deb
+* Display a short README à la Github's rendering of README.md
+* Show plugin popularity/trustworthiness via download stats
+* Plugin management (list/install/uninstall) via a command like
+  foreman-rake/hammer - does it even make sense for us to do?
+
+# Detailed design
+[design]: #detailed-design
+
+A Foreman Forge page for plugin would provide all the information about that
+plugin at a glance.
+
+```
++-----------------------------------------------------------------------------+
+| Foreman Forge                                                               |
++------------------------------------------------+----------------------------+
+|                                                |                            |
+|  foreman_my_plugin                             | [Downloads] 1,337          |
+|                                                | [Rating] ****              |
+|  [summary] My plugin for Foreman adds support  +----------------------------+
+|  for my functionality so that users can do foo.|                            |
++------------------------------------------------+ [gem install plugin]       |
+|                                                |                            |
+|  [README.md] This section automatically reads  | [yum install plugin]       |
+|  contents of README.md and displays the docs.  |                            |
+|  README.md is the place where plugin authors   | [apt-get install plugin]   |
+|  are expected to write installation steps and  |                            |
+|  other relevant documentation.                 | [github.com/user/plugin]   |
+|                                                |                            |
++------------------------------------------------+----------------------------+
+```
+
+## Data sources
+
+Foreman plugins are implemented as Ruby Gems. Therefore, a lot of metadata is
+already available in the form of Gemspec attributes.
+
+* foreman_plugin_name
+* Plugin summary
+* Author
+
+Additionally, more metadata is available from RubyGems API
+(example:
+[foreman_remote_execution](https://rubygems.org/api/v1/gems/foreman_remote_execution.json))
+
+* Downloads
+* Downloads for current version
+* Github URL
+* Documentation URL
+
+## Web application
+
+Depending on how rich we want the Forge to be, one of the following approaches
+may be taken.
+
+### Jekyll based static Forge
+
+We're already using Jekyll for building a static website with much success. A
+Forge based on Jekyll would have:
+
+* A master plugin list in JSON/YAML format. Plugins can be added/remove from the
+  catalog by sending a PR against this file.
+
```

Comment:
What if the master list was on our side, but the plugin details yaml was in the git repo fully under the author's control? Our script would periodically scan the the plugins repos from the list for changes and re-generate the jekyll page.